src/adc.rs

```diff
@@ -0,0 +1,171 @@
+//! TODO
+
+use embedded_hal::adc::{Channel, OneShot};
+
+use crate::{
+    init_state, pac, swm,
+    syscon::{self, clocksource::AdcClock},
+};
+
+/// Interface to the ADC peripheral
+///
+/// Controls the ADC.  Use [`Peripherals`] to gain access to an instance of
+/// this struct.
+///
+/// Please refer to the [module documentation] for more information.
+///
+/// [`Peripherals`]: ../struct.Peripherals.html
+/// [module documentation]: index.html
+pub struct ADC<State = init_state::Enabled> {
+    adc: pac::ADC0,
+    _state: State,
+}
+
+impl ADC<init_state::Disabled> {
+    pub(crate) fn new(adc: pac::ADC0) -> Self {
+        Self {
+            adc,
+            _state: init_state::Disabled,
+        }
+    }
+    /// Enable the ADC
+    ///
+    /// This method is only available, if `ADC` is in the [`Disabled`] state.
+    /// Code that attempts to call this method when the peripheral is already
+    /// enabled will not compile.
+    ///
+    /// Consumes this instance of `ADC` and returns another instance that has
+    /// its `State` type parameter set to [`Enabled`].
+    ///
+    /// # Limitations
+    ///
+    /// This assumes that the clock is running at 500 kHz for the calibration
```

Comment:
Might be worth specifying which clock that is. At least it's not obvious to me, not having touched an LPC800 ADC in a while.

Reply:
I think it's fine to remove this, since it's already documented in clocksource.rs, which is more clear